### PR TITLE
gateway: bump gateway image to use the latest available

### PR DESCRIPTION
**Pull request by cpanato**
## Description
Bump the gateway image to use the latest available and also support some features like https://github.com/openfaas/faas-netes/pull/772

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

/assign @alexellis

### CODE REVIEW

File: yaml/gateway-dep.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ spec:
             requests:
               cpu: 50m
               memory: 120Mi
-        image: ghcr.io/openfaas/gateway:0.20.11
+        image: ghcr.io/openfaas/gateway:0.20.12
```
Comment (alexellis) [inline yaml/gateway-dep.yml:36]: FYI this file is generated

**Review by alexellis — DISMISSED** [review]
Approved

**Comment by alexellis** [comment]
Could you also bump faas-netes in the same PR please? https://github.com/openfaas/faas-netes/releases/tag/0.13.2

File: yaml/gateway-dep.yml (hunk 1 above)
Comment (cpanato) [reply yaml/gateway-dep.yml:36]: how do I generate/update this file?

Comment (cpanato) [reply yaml/gateway-dep.yml:36]: found it

**Comment by cpanato** [comment]
@alexellis PTAL

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**